Lombiq.HelpfulLibraries.AspNetCore/Middlewares/DeferredTaskMiddleware.cs

```diff
@@ -15,15 +16,17 @@ public class DeferredTaskMiddleware
         HttpContext context,
         IEnumerable<IDeferredTask> deferredTasks)
     {
-        foreach (var deferredTask in deferredTasks)
+        var deferredTasksList = deferredTasks.ToList();
```

Comment:
Prefer `enumerable.AsList()` whenever possible. Unless you know for sure that the enumerable is something weird like a query, it's very likely to already be an `IList<T>` instance like a list or array.  This way you avoid an unnecessary copy; `enumerable.AsList()` is basically free.

![image](https://user-images.githubusercontent.com/4007293/212082398-9d1cea9d-b678-49c6-ac23-75f24bce048a.png)
(looks like MS dependency injection uses an array for service collections. This is not surprising, because the collection is fixed by `services.BuildServiceProvider()`.)

Reply:
I wasn't aware of `AsList()`, thanks.